src/components/HeaderNavMenu.jsx

```diff
@@ -30,8 +52,27 @@ function HeaderNavMenu() {
           </MuiLink>
         </Grid>
       ))}
-      <Grid item>
-        <LoginButton className={classes.menuItem} />
+      <Grid item className={classes.avatarMenu}>
+        {(loading && <null />) ||
+          (isSignedIn ? (
+            <ProfileMenu
+              signOut={signUserOut}
+              name={user.displayName}
+              email={user.email}
+              photo={user.photoURL}
+            />
+          ) : (
+            <Button
+              variant="contained"
+              color="primary"
+              component={Link}
+              to={{
+                state: { showLoginDialog: true },
+              }}
+              replace>
+              Login
+            </Button>
```

Comment:
You can copy this to the other place you used `LoginButton`.

Reply:
Ok it worked but now I noticed that when the user is logged in the change to the avatar does not happen. Im assuming the logic for displaying the profile avatar or login button is used there as well to get it to work

Reply:
The logic switch between Avatar and LoginButton on the `ScholarshipsMadeSimple` is implemented and fully functioning.